tests/test_datasets/test_datasets.py

```diff
@@ -874,3 +874,191 @@ def test_extra_repr(self):
+        dataset = dataset_class(**cfg)
+        self.assertDictEqual(dataset.metainfo, metainfo)
+        self.assertDictEqual(dataset.CLASSES, {
+            'gender': ['a', 'b'],
+            'wear': ['c', 'd', 'e']
+        })
+
+        # Test invalid metainfo
+        cfg = {**self.DEFAULT_ARGS, 'metainfo': ['a', 'b']}
+        with self.assertRaisesRegex(AssertionError, "got <class 'list'>"):
+            dataset_class(**cfg)
+
+        # Test annotation file without metainfo
+        tmpdir = tempfile.TemporaryDirectory()
+        new_annotation = mmcv.load(dataset.ann_file)
+        del new_annotation['metainfo']
+        new_ann_file = osp.abspath(osp.join(tmpdir.name, 'ann_file.json'))
+        mmcv.dump(new_annotation, new_ann_file)
+        cfg = {**self.DEFAULT_ARGS, 'ann_file': new_ann_file}
+        with self.assertRaisesRegex(AssertionError, 'specify the `tasks`'):
+            dataset_class(**cfg)
+
+        # Test wrong task type
+        cfg['metainfo'] = {'tasks': [['a', 'b'], ['d', 'e']]}
+        with self.assertRaisesRegex(AssertionError, 'should be a dict'):
+            dataset_class(**cfg)
+
+        # Test incomplete metainfo
+        cfg['metainfo'] = {
+            'tasks': [{
+                'name': 'gender',
+                'categories': ['a', 'b'],
+            }]
+        }
+        with self.assertRaisesRegex(AssertionError, 'Missing "type"'):
+            dataset_class(**cfg)
+
+        tmpdir.cleanup()
+
+    def test_data_root(self):
+        dataset_class = DATASETS.get(self.DATASET_TYPE)
+
+        # Test relative ann_file
+        cfg = {**self.DEFAULT_ARGS, 'ann_file': 'multi-task.json'}
+        dataset = dataset_class(**cfg)
+        self.assertEqual(dataset.data_root, ASSETS_ROOT)
+        self.assertEqual(dataset.ann_file,
+                         osp.join(dataset.data_root, 'multi-task.json'))
+
+        # Test relative data_prefix
+        cfg = {**self.DEFAULT_ARGS, 'data_prefix': 'train'}
+        dataset = dataset_class(**cfg)
+        self.assertEqual(dataset.data_prefix,
+                         osp.join(dataset.data_root, 'train'))
+
+        # Test no data_prefix
+        cfg = {**self.DEFAULT_ARGS, 'data_prefix': None}
+        dataset = dataset_class(**cfg)
+        self.assertEqual(dataset.data_prefix, dataset.data_root)
+
+    def test_parse_data_info(self):
+        dataset_class = DATASETS.get(self.DATASET_TYPE)
+        dataset = dataset_class(**self.DEFAULT_ARGS)
+
+        data = dataset.parse_data_info({
+            'img_path': 'a.jpg',
+            "gt_label":{"gender": 0}
+        })
+        self.assertDictContainsSubset(
+            {
+                'img_path': ASSETS_ROOT+'/a.jpg',
+                'gender': 0
+            }, data)
+        np.testing.assert_equal(data['gender'],0)
+
+
+        # Test invalid type
+        with self.assertRaisesRegex(AssertionError, "got <class 'str'>"):
+            dataset.parse_data_info('hi')
```

Comment:
This should not be done by the MultiTaskDataset, the idea is to have something very generic